`storage`: replace a `segment`'s `batch_cache_index` after compaction

WillemKauf opened the pull request:
Previously, a `segment` would release its `batch_cache_index` after being compacted (either via self compaction or sliding window compaction) and would never replace it, barring a restart.

This commit changes the behavior to allow segments to reset their `batch_cache_index` instead of releasing it after being compacted.



## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

### Improvements

* Fixes sub-optimal behavior in which a `segment`'s `batch_cache_index` would be released after compaction and never re-assigned.

WillemKauf committed:
`storage`: add `batch_cache_index::reset()`

A new API that allows the caller to fully reset the `batch_cache_index`
in a `segment`, leaving it in a clean and reusable state.

secpanda commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/devex-c3w/pr-checks/73312339-9c17-4c4b-a9f8-af6e719c8355?refs=3d82ccde1a653e649190f9e114061b305b90952c&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/devex-c3w/pr-checks/73312339-9c17-4c4b-a9f8-af6e719c8355/license?refs=3d82ccde1a653e649190f9e114061b305b90952c&source=prChecksComment)

secpanda commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/devprod/pr-checks/b10615d3-d5eb-4fb0-97c3-3bd06e80c3d3?refs=3d82ccde1a653e649190f9e114061b305b90952c&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/devprod/pr-checks/b10615d3-d5eb-4fb0-97c3-3bd06e80c3d3/license?refs=3d82ccde1a653e649190f9e114061b305b90952c&source=prChecksComment)

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/65058-->
<!--9cf28af2-97f7-43b3-ad7c-263623e748e1-->
<!--d278f288-8478-4b43-83f7-86f0cd03176c-->
<!--title-->
## Retry command for [Build#65058](https://buildkite.com/redpanda/redpanda/builds/65058)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```

/ci-repeat 1
tests/rptest/tests/basic_kafka_compat_test.py::TxKafkaCompatTest.test_concurrent_producer_ids@{"metadata_quorum":"COMBINED_KRAFT"}
tests/rptest/tests/tiered_storage_pause_test.py::TestTieredStoragePause.test_safe_pause_resume@{"allow_gaps_cluster_level":true,"allow_gaps_topic_level":true}
```
<!--/body-->

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/65071-->
<!--b737a400-0ccb-4e73-b30a-1438687b3e93-->
<!--afbeb48e-75cf-48c6-81ae-423a2ebd80b7-->
<!--a4193762-b06b-41a1-bf57-cd5f64cbeebf-->
<!--0e470a18-4b85-4964-a550-7106c12a71a7-->
<!--c75240cb-d695-4964-ae1a-e26973768c09-->
<!--6855feb4-3781-45ba-81c8-1a6d842e89cd-->
<!--title-->
## Retry command for [Build#65071](https://buildkite.com/redpanda/redpanda/builds/65071)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_migrated_topic_data_integrity@{"params":{"cancellation":{"dir":"in","stage":"preparing"},"use_alias":true},"transfer_leadership":true}
tests/rptest/tests/partition_move_interruption_test.py::PartitionMoveInterruption.test_cancelling_partition_move_x_core@{"compacted":false,"recovery":"restart_recovery","replication_factor":1,"unclean_abort":true}
tests/rptest/tests/partition_balancer_test.py::PartitionBalancerTest.test_full_nodes
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_migrated_topic_data_integrity@{"params":{"cancellation":{"dir":"in","stage":"preparing"},"use_alias":false},"transfer_leadership":true}
tests/rptest/tests/random_node_operations_test.py::RandomNodeOperationsTest.test_node_operations@{"cloud_storage_type":1,"enable_failures":false,"mixed_versions":false,"with_chunked_compaction":true,"with_iceberg":true,"with_tiered_storage":false}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--7e390338-41c9-4649-b23c-733b9f5ea235-->
<!--TEST RESULTS-->
<!--de3b19f1-1f64-4a5b-b829-cd663ff3f9fd-->
<!--TEST RESULTS-->
<!--b940dc44-0197-4ccd-a2bc-aaa21350a8ac-->
<!--TEST RESULTS-->
<!--aa9e7eb2-78e5-4dcd-94f1-a723b8cc1e24-->
<!--TEST RESULTS-->
<!--faf1f1a2-e825-4f5e-afd3-0141ebd4d1b9-->
<!--TEST RESULTS-->
<!--e51adbd7-539d-447c-b6f7-22493ce284e9-->
<!--TEST RESULTS-->
<!--c0f348f1-909d-4d38-b9e1-31de230de94f-->
<!--TEST RESULTS-->
<!--c6bde2fe-5a0a-4ae2-95b1-439a935a1cb0-->
<!--TEST RESULTS-->
<!--808bdf8d-4cc2-4888-b722-fae845674ad0-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65071">build#65071</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=True.params=.cancellation.dir.in.stage.preparing.use_alias.True | ducktape | https://buildkite.com/redpanda/redpanda/builds/65071#01966b00-5f6a-47e3-8966-91e5c1c23315 | FLAKY | 15/21 |
| rptest.tests.partition_balancer_test.PartitionBalancerTest.test_full_nodes | ducktape | https://buildkite.com/redpanda/redpanda/builds/65071#01966afc-00cd-4f0e-a0a6-70441ae1ae7a | FLAKY | 20/21 |
| rptest.tests.partition_move_interruption_test.PartitionMoveInterruption.test_cancelling_partition_move_x_core.replication_factor=1.unclean_abort=True.recovery=restart_recovery.compacted=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/65071#01966b00-5f6d-4d0e-b1be-217d1a9b3874 | FLAKY | 20/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65180">build#65180</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-----------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_conflicting_names | ducktape | https://buildkite.com/redpanda/redpanda/builds/65180#01967cdf-0345-4c27-9511-f1b03a3d8e52 | FLAKY | 3/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65195">build#65195</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.availability_test.AvailabilityTests.test_availability_when_one_node_failed | ducktape | https://buildkite.com/redpanda/redpanda/builds/65195#01967e8f-8136-4fb9-ba7c-1af12315ff95 | FLAKY | 20/21 |
| rptest.tests.cloud_storage_timing_stress_test.CloudStorageTimingStressTest.test_cloud_storage_with_partition_moves.cleanup_policy=compact.delete | ducktape | https://buildkite.com/redpanda/redpanda/builds/65195#01967e93-8318-4318-a68a-b14f00d6da20 | FLAKY | 19/21 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_conflicting_names | ducktape | https://buildkite.com/redpanda/redpanda/builds/65195#01967e8f-8138-4998-bee8-4373ed35d6bb | FLAKY | 4/21 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_conflicting_names | ducktape | https://buildkite.com/redpanda/redpanda/builds/65195#01967e93-831a-445d-b384-cc183ce78464 | FLAKY | 7/21 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_fast_node_addition | ducktape | https://buildkite.com/redpanda/redpanda/builds/65195#01967e93-831a-445d-b384-cc183ce78464 | FLAKY | 20/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65211">build#65211</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-----------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_conflicting_names | ducktape | https://buildkite.com/redpanda/redpanda/builds/65211#01967fc0-c403-4dfd-b37b-c2523ddb0a0c | FLAKY | 8/21 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_conflicting_names | ducktape | https://buildkite.com/redpanda/redpanda/builds/65211#01967fd3-295f-4125-9c6d-7810051ce6bd | FLAKY | 8/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65234">build#65234</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-----------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_conflicting_names | ducktape | https://buildkite.com/redpanda/redpanda/builds/65234#019681f8-3a7f-47f6-982b-883b0c5c0ab2 | FLAKY | 5/21 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_conflicting_names | ducktape | https://buildkite.com/redpanda/redpanda/builds/65234#019681fc-3b14-4912-9aa2-b26cb3dc5b1b | FLAKY | 8/21 |
| rptest.tests.simple_e2e_test.SimpleEndToEndTest.test_consumer_interruption | ducktape | https://buildkite.com/redpanda/redpanda/builds/65234#019681f8-3a7e-456b-b13f-918035f2f047 | FLAKY | 18/21 |
| rptest.tests.simple_e2e_test.SimpleEndToEndTest.test_consumer_interruption | ducktape | https://buildkite.com/redpanda/redpanda/builds/65234#019681fc-3b14-4912-9aa2-b26cb3dc5b1b | FLAKY | 7/21 |
| rptest.tests.upgrade_test.UpgradeBackToBackTest.test_upgrade_with_all_workloads.single_upgrade=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/65234#019681f8-3a7c-4da7-965e-4b64b75f2b93 | FLAKY | 12/21 |
| rptest.tests.upgrade_test.UpgradeBackToBackTest.test_upgrade_with_all_workloads.single_upgrade=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/65234#019681fc-3b14-4912-9aa2-b26cb3dc5b1b | FLAKY | 15/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65294">build#65294</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-----------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_conflicting_names | ducktape | https://buildkite.com/redpanda/redpanda/builds/65294#019686c1-c3c5-4a62-96c1-56ea37c287e1 | FLAKY | 5/21 |
| rptest.tests.simple_e2e_test.SimpleEndToEndTest.test_consumer_interruption | ducktape | https://buildkite.com/redpanda/redpanda/builds/65294#019686c1-c3ca-417f-a40e-ce8267e9069e | FLAKY | 6/21 |
| rptest.tests.upgrade_test.UpgradeBackToBackTest.test_upgrade_with_all_workloads.single_upgrade=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/65294#019686c1-c3c7-48ac-8a4a-17ad9ec1007e | FLAKY | 13/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65310">build#65310</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:-----------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_conflicting_names | ducktape | https://buildkite.com/redpanda/redpanda/builds/65310#0196876c-5138-49bf-a7e2-a47d1d7225bc | FLAKY | 10/21 |
| rptest.tests.simple_e2e_test.SimpleEndToEndTest.test_consumer_interruption | ducktape | https://buildkite.com/redpanda/redpanda/builds/65310#0196876c-5138-4aa8-b55a-b8f3d955c764 | FLAKY | 7/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65314">build#65314</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:---------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.simple_e2e_test.SimpleEndToEndTest.test_consumer_interruption | ducktape | https://buildkite.com/redpanda/redpanda/builds/65314#01968797-b7c6-4427-ba8f-1a5da702c339 | FLAKY | 7/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/65331">build#65331</a></summary>


| test_id                                                                    | test_kind   | job_url                                                                                   | test_status   | passed   |
|:---------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.simple_e2e_test.SimpleEndToEndTest.test_consumer_interruption | ducktape    | https://buildkite.com/redpanda/redpanda/builds/65331#01968843-9631-4172-9e87-c2d057d5bffb | FLAKY         | 6/21     |
</details>
<!--/body-->

nvartolomei commented:
Rebased from the GitHub UI to get rid of already merged commits from the diff.

nvartolomei commented on src/v/storage/log_manager.h:
```diff
@@ -206,6 +206,8 @@ class log_manager {
     ss::future<> start();
     ss::future<> stop();
 
+    std::optional<batch_cache_index> create_cache(with_cache);
```
What if instead of making this public we make use of `batch_cache_index::clear_async`? We don't need to release the batch cache index. Moreover, releasing is a sync operation which led to reactor stalls in the past and that's why we have clear_async now so we get an additional win! https://github.com/redpanda-data/redpanda/pull/23074/commits/4272600839d6658c56dd6c29956eeb9794740641

nvartolomei replied:
~~The async method on the other hand will prevent reading from the segment because the cache index will be locked 🤔~~

~~Not worse than locking the reactor 🤔~~

We hold a write lock anyway.

WillemKauf replied:
`clear_async()` doesn't re-initialize `_small_batches_range` to `nullptr`, should it?

The only previous use of `clear_async()` is in `segment::do_close()`, after which the batch cache won't be re-used- i.e, if we make this change, this will be the first time the `batche_cache_index` is going to be re-used after a `clear_async()` call, so we now need to consider clearing all state here.

nvartolomei replied:
> clear_async() doesn't re-initialize _small_batches_range to nullptr, should it?

From reading the code it seems that doing that won't hurt but not doing it won't have any correctness implications. It is a weak ptr and by the time clear_async is done that pointer should have been deleted already by the calls to `_cache->evict` based on index entries.

> The only previous use of clear_async() is in segment::do_close(), after which the batch cache won't be re-used- i.e, if we make this change, this will be the first time the batche_cache_index is going to be re-used after a clear_async() call, so we now need to consider clearing all state here.

Yes.

WillemKauf replied:
I am considering adding a `batch_cache_index::reset()` function  which will invoke both of these and make the future reuse clear. :)

WillemKauf replied:
Added

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/65180-->
<!--03146f42-8740-42c8-886d-f5681e0137b8-->
<!--title-->
## Retry command for [Build#65180](https://buildkite.com/redpanda/redpanda/builds/65180)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_conflicting_names
```
<!--/body-->

WillemKauf committed:
`storage`: `reset`, not `replace`, the `batch_cache_index`

Previously, a `segment` would release its `batch_cache_index` after being
compacted (either via self compaction or sliding window compaction) and
would never replace it, barring a restart.

This commit changes the behavior to allow segments to reset their
`batch_cache_index` instead of releasing it after being compacted.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/65195-->
<!--1161f1b3-8da6-424f-a031-b887bee46b86-->
<!--ac765ec0-acd9-4927-ad7f-a923c5a53feb-->
<!--a8949c44-b8f6-4cbb-8c4a-006d855ab726-->
<!--8115ef8c-7fc8-4528-a96a-f8e3c2b17dce-->
<!--title-->
## Retry command for [Build#65195](https://buildkite.com/redpanda/redpanda/builds/65195)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/availability_test.py::AvailabilityTests.test_availability_when_one_node_failed
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_conflicting_names
tests/rptest/tests/scaling_up_test.py::ScalingUpTest.test_fast_node_addition
tests/rptest/tests/cloud_storage_timing_stress_test.py::CloudStorageTimingStressTest.test_cloud_storage_with_partition_moves@{"cleanup_policy":"compact,delete"}
```
<!--/body-->

WillemKauf commented:
/ci-repeat 1
tests/rptest/tests/availability_test.py::AvailabilityTests.test_availability_when_one_node_failed
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_conflicting_names
tests/rptest/tests/scaling_up_test.py::ScalingUpTest.test_fast_node_addition
tests/rptest/tests/cloud_storage_timing_stress_test.py::CloudStorageTimingStressTest.test_cloud_storage_with_partition_moves@{"cleanup_policy":"compact,delete"}

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/65211-->
<!--69a3811b-aac5-4c61-bcaf-db85bd6a0172-->
<!--50f9718b-27d7-4ab0-ba1d-0b2c78aca914-->
<!--title-->
## Retry command for [Build#65211](https://buildkite.com/redpanda/redpanda/builds/65211)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```

/ci-repeat 1
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_conflicting_names
```
<!--/body-->

nvartolomei commented on src/v/storage/batch_cache.h:
```diff
@@ -622,6 +622,12 @@ class batch_cache_index {
     }
 
+    // Leaves the batch_cache_index in a fully clean, re-usable state.
+    ss::future<> reset() {
```
needs to acquire the lock otherwise there is a race between clear_async and _small_batches_range?

might not be problematic in practice but i'd prefer we have all this section under lock

WillemKauf replied:
Thanks for the callout, added `clear_async_unlocked()` to support this.

nvartolomei commented on src/v/storage/segment.h:
```diff
@@ -222,6 +222,7 @@ class segment {
     bool has_compactible_offsets(const compaction_config& cfg) const;
 
     void release_batch_cache_index() { _cache.reset(); }
```
dead code now?

WillemKauf replied:
Removed dead code.

WillemKauf committed:
`storage`: add `batch_cache_index::clear_async_unlocked()`

This is an unlocked version of `clear_async()` that assumes a
`lock_guard` constructed from `*this` is already held from the call
site.

This will be used in a future commit.

WillemKauf committed:
`storage`: remove dead code `segment::release_batch_cache_index()`

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/65234-->
<!--90ce4e2c-7540-42f0-807f-befb3786d440-->
<!--28b9ae6b-594e-4f7a-b9d4-a95967b4104d-->
<!--7c45e0d9-902d-4e56-ae34-5c4253b61b20-->
<!--title-->
## Retry command for [Build#65234](https://buildkite.com/redpanda/redpanda/builds/65234)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/upgrade_test.py::UpgradeBackToBackTest.test_upgrade_with_all_workloads@{"single_upgrade":false}
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_conflicting_names
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_consumer_interruption
```
<!--/body-->

WillemKauf commented:
/ci-repeat 1
skip-redpanda-build
skip-units
tests/rptest/tests/upgrade_test.py::UpgradeBackToBackTest.test_upgrade_with_all_workloads@{"single_upgrade":false}
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_conflicting_names
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_consumer_interruption

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/65294-->
<!--6416f8d3-bc3c-47e9-886e-f6182d4bbaa0-->
<!--7e0e839d-a5ed-4660-ad86-3214b5339db6-->
<!--a1d00b79-47ef-4100-846d-81a82693c53c-->
<!--title-->
## Retry command for [Build#65294](https://buildkite.com/redpanda/redpanda/builds/65294)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_consumer_interruption
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_conflicting_names
tests/rptest/tests/upgrade_test.py::UpgradeBackToBackTest.test_upgrade_with_all_workloads@{"single_upgrade":false}
```
<!--/body-->

WillemKauf commented:
/ci-repeat 1
skip-redpanda-build
skip-units
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_consumer_interruption
tests/rptest/tests/data_migrations_api_test.py::DataMigrationsApiTest.test_conflicting_names
tests/rptest/tests/upgrade_test.py::UpgradeBackToBackTest.test_upgrade_with_all_workloads@{"single_upgrade":false}

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/65310-->
<!--db60dc8d-2e60-4ee8-8428-ea96bc025bee-->
<!--title-->
## Retry command for [Build#65310](https://buildkite.com/redpanda/redpanda/builds/65310)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_consumer_interruption
```
<!--/body-->

WillemKauf commented:
/ci-repeat 1
skip-redpanda-build
skip-units
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_consumer_interruption

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/65314-->
<!--c8291fa5-1fbf-43b8-9697-b22f9a428fd2-->
<!--title-->
## Retry command for [Build#65314](https://buildkite.com/redpanda/redpanda/builds/65314)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_consumer_interruption
```
<!--/body-->

WillemKauf commented:
/ci-repeat 1
skip-redpanda-build
skip-units
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_consumer_interruption

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/65331-->
<!--a7894a52-20e9-435c-92e1-cd8091a36308-->
<!--title-->
## Retry command for [Build#65331](https://buildkite.com/redpanda/redpanda/builds/65331)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_consumer_interruption
```
<!--/body-->

WillemKauf commented:
/ci-repeat 1
skip-redpanda-build
skip-units
tests/rptest/tests/simple_e2e_test.py::SimpleEndToEndTest.test_consumer_interruption

dotnwat reviewed (commented):
missed this when it flew by. nice change
